src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -103,25 +105,21 @@ OverlayImpl::Timer::on_timer(error_code ec)
     if ((overlay_.timer_count_ % Tuning::checkIdlePeers) == 0)
         overlay_.deleteIdlePeers();
 
-    timer_.expires_from_now(std::chrono::seconds(1));
-    timer_.async_wait(overlay_.strand_.wrap(std::bind(
-        &Timer::on_timer, shared_from_this(), std::placeholders::_1)));
+    start();
```

Comment:
That's a little weird. I'd expect the `start` function to only be called once.

Reply:
Think of it this way and let me know if that changes your expectation:

`Timer` is a state machine. `start` is a transition that enters the "waiting" state. It doesn't care what state it's coming from. It can be called from any state, including the "just constructed" state and the "notified" state.